lms/djangoapps/support/static/support/js/models/enrollment.js

```diff
@@ -0,0 +1,24 @@
+(function (define) {
+    'use strict';
+    define(['backbone', 'underscore'], function (Backbone, _) {
+        return Backbone.Model.extend({
+            updateEnrollment: function (new_mode, reason) {
+                return $.ajax({
+                    url: this.url(),
+                    type: 'POST',
+                    contentType: 'application/json',
+                    data: JSON.stringify({
+                        course_id: this.get('course_id'),
+                        new_mode: new_mode,
+                        old_mode: this.get('mode'),
+                        reason: reason
+                    }),
+                    success: _.bind(function (response) {
+                        this.set('manual_enrollment', response);
+                        this.set('mode', new_mode);
+                    }, this)
```

Comment:
There's no error handling necessary on the model itself, but since it return the result of `$.ajax` error handlers can be attached by the calling code. This happens [in the modal view.](https://github.com/edx/edx-platform/pull/10966/files#diff-d8d718b7fbe13c7b74307820f77431c3R54)